images/virtualization-artifact/pkg/monitoring/metrics/util/labels.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+var (
+	invalidLabelCharRE = regexp.MustCompile(`[^a-zA-Z0-9_]`)
+	matchAllCap        = regexp.MustCompile("([a-z0-9])([A-Z])")
+)
+
+type SkipLabel func(key, value string) bool
+
+func WrapPrometheusLabels(labels map[string]string, prefix string, skip SkipLabel) map[string]string {
+	wrapLabels := make(map[string]string, len(labels))
+	conflicts := make(map[string]int, len(labels))
+
+	for k, v := range labels {
+		if skip != nil && skip(k, v) {
+			continue
+		}
+		labelKey := labelName(prefix, k)
+		if conflictCount, ok := conflicts[labelKey]; ok {
+			if conflictCount == 1 {
+				// this is the first conflict for the label,
+				// so we have to go back and rename the initial label that we've already added
+
+				value := wrapLabels[labelKey]
+				delete(wrapLabels, labelKey)
+				wrapLabels[labelConflictSuffix(labelKey, conflictCount)] = value
+			}
```

Comment:
Here we save the value from the first conflicting label, so labels "my-vm-name" and "my/vm-name" will produce "prefixed_my_vm_label_conflict1" with unpredictable value.
I think we need to add sorting for map keys to stabilize metric label values between scrapes.